src/Services/ContactsListService.php

```diff
@@ -0,0 +1,219 @@
+        }
+        $response = $this->_exec($listId, $contact);
+        if (!$response->success()) {
+            $this->throwError("ContactsListManager:sub() failed", $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * unsubscribe a Contact from listId
+     * @param string $listId
+     * @param Contact $contact
+     */
+    public function unsubscribe($listId, Contact $contact)
+    {
+        $contact->setAction(Contact::ACTION_UNSUB);
+        $response = $this->_exec($listId, $contact);
+        if (!$response->success()) {
+            $this->throwError("ContactsListManager:unsub() failed", $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Delete a Contact from listId
+     * @param string $listId
+     * @param Contact $contact
+     */
+    public function delete($listId, Contact $contact)
+    {
+        $contact->setAction(Contact::ACTION_REMOVE);
+        $response = $this->_exec($listId, $contact);
+        if (!$response->success()) {
+            $this->throwError("ContactsListManager:delete() failed", $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Change email a Contact
+     * @param string $listId
+     * @param Contact $contact
+     * @param string $oldEmail
+     */
+    public function changeEmail($listId, Contact $contact, $oldEmail)
+    {
+        // get old contact properties
+        $response = $this->mailjet->get(Resources::$Contactdata, ['id' => $oldEmail]);
+        if (!$response->success()) {
+            $this->throwError("ContactsListManager:changeEmail() failed", $response);
+        }
+
+        // copy contact properties
+        $oldContactData = $response->getData();
+        if (isset($oldContactData[0])) {
+            $contact->setProperties($oldContactData[0]['Data']);
+        }
+
+        // add new contact
+        $contact->setAction(Contact::ACTION_ADDFORCE);
+        $response = $this->_exec($listId, $contact);
+        if (!$response->success()) {
+            $this->throwError("ContactsListManager:changeEmail() failed", $response);
+        }
+
+        // remove old
+        $oldContact = new Contact($oldEmail);
+        $oldContact->setAction(Contact::ACTION_REMOVE);
+        $response = $this->_exec($listId, $oldContact);
+        if (!$response->success()) {
+            $this->throwError("ContactsListManager:changeEmail() failed", $response);
+        }
+
+        return $response->getData();
+    }
+
+    /**
+     * Manage Many Contacts to List
+     * https://dev.mailjet.com/email-api/v3/contactslist-managemanycontacts/
+     * @param  ContactsList $contactsList
+     * @return array
+     */
+    public function manageManyContactsList(ContactsList $contactsList)
+    {
+        $batchResults = [];
+        // we send multiple smaller requests instead of a bigger one
+        $contactChunks = array_chunk($contactsList->getContacts(), self::CONTACT_BATCH_SIZE);
```

Comment:
The API will always return up to 1000 (default value for `self::CONTACT_BATCH_SIZE`). You then will always send one request. Plus I don't see pagination to handle lists bigger than 1000s contacts?